### PR TITLE
Add Auth service usage by the Framework

### PR DESCRIPTION
**Pull request by maza4aka**
According to their documentation, this is all that is needed to bind our
Auth service to the Authentication procedure.

Users` registration has nothing to do with this - registartion happens
without JWT usage, somewhere else in rest module.

### CODE REVIEW

File: src/main/java/edu/kpi/testcourse/rest/AuthenticationProviderUserPassword.java [hunk 1]
```diff
@@ -26,11 +29,12 @@
       @Nullable HttpRequest<?> httpRequest,
       AuthenticationRequest<?, ?> authenticationRequest
   ) {
-    // TODO Here you need to implement an actual authentication (ensure that the user is registered
-    //  and password is OK)
     return Flowable.create(emitter -> {
-      if (authenticationRequest.getIdentity().equals("sherlock")
-          && authenticationRequest.getSecret().equals("password")) {
+      Auth auth = Main.getAuth();
+      var username = (String) authenticationRequest.getIdentity();
```
[the remark this answers is not in the document]
Comment (maza4aka) [reply src/main/java/edu/kpi/testcourse/rest/AuthenticationProviderUserPassword.java:34]: var lets me avoid writing type two times in the same statement (I'm casting function call result to String explicitly).

[the remark this answers is not in the document]
Comment (maza4aka) [reply src/main/java/edu/kpi/testcourse/rest/AuthenticationProviderUserPassword.java:34]: bruh, just let me merge c:

[the remark this answers is not in the document]
Comment (AsyaAndKo) [reply src/main/java/edu/kpi/testcourse/rest/AuthenticationProviderUserPassword.java:34]: But, while we're in jab(v)a, imho it's better to use good old strict typisation

[the remark this answers is not in the document]
Comment (maza4aka) [reply src/main/java/edu/kpi/testcourse/rest/AuthenticationProviderUserPassword.java:34]: bruh

ok, I'll fix it

[the remark this answers is not in the document]
Comment (AsyaAndKo) [reply src/main/java/edu/kpi/testcourse/rest/AuthenticationProviderUserPassword.java]: bruh

**Review by AsyaAndKo — APPROVED** [review]

**Pull request merged**